src/libraries/System.Private.CoreLib/src/System/Environment.Variables.Windows.cs

```diff
@@ -92,44 +79,43 @@ public static unsafe IDictionary GetEnvironmentVariables()
                 // environment block (as =C:=pwd) and the program's exit code is
                 // as well (=ExitCode=00000000).
 
+                // Search for terminating \0\0 (two unicode \0's).
+                // IndexOf processes memory in aligned chunks, and thus it won't crash even if it accesses memory beyond the null terminator.
```

Comment:
IndexOf searches for the first char (ie \0 in this case) and then compares the rest.

It should better to just search for the next terminating zero inside the loop, using `wcslen`

Reply:
Sounds good.

Reply:
Just to clarify,  did you mean:
a) replace the IndexOf with my own loop that uses wcslen + checking the next character,  or
b) delete the IndexOf, and change the main loop to use wcslen on each iteration to find the next terminator
?

Reply:
Yep (b)

Reply:
That cleans things up nicely, and is a tad faster.